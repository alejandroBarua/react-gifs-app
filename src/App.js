import React, { useState } from 'react';
import PropTypes from 'prop-types';

import AddCategory from './components/AddCategory';
import GifGrid from './components/GifGrid';

const App = ({ title }) => {

	const [categories, setCategories] = useState(['Rick and Morty', 'Game of Thrones']);

	//const handleAdd = () => setCategories(['Dragon Ball', ...categories]);
	
	return (

		<> 
			<h1>{title}</h1>

			<AddCategory 
				setCategories={setCategories} 
			/>

			<hr />
			{/* <button onClick={handleAdd}>Add</button> */}

			<ol>
				{
					categories.map(category => <GifGrid key={category} category={category} />)
				}
			</ol>
		</>

	);
} 

App.propTypes = {
	title: PropTypes.string.isRequired 
}


export default App; 