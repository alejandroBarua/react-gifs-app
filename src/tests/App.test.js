import { shallow } from 'enzyme';
import React from 'react';
import App from '../App';

describe('App component', () => {
	
	test('render component', () => {
		const title = "Hello World";
		const wrapper = shallow(<App title={title}/>);

		expect(wrapper).toMatchSnapshot();
	})
	
})
