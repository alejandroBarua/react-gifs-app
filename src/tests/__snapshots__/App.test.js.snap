// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component render component 1`] = `
<Fragment>
  <header>
    <div
      class="container"
    >
      <h1>
        Hello World
      </h1>
      <AddCategory
        setCategories={[Function]}
      />
    </div>
  </header>
  <div
    class="container"
  >
    <ol>
      <GifGrid
        category="Rick and Morty"
        key="Rick and Morty"
      />
    </ol>
  </div>
</Fragment>
`;
